File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cs
public class Solution {
    public int FindMin(int[] nums) {
        var res = nums[0];

        var l = 0;
        var r = nums.Length - 1;

        while (l <= r)
        {
            if (nums[l] < nums[r])
            {
                res = Math.Min(res, nums[l]);
                break;
            }

            int mid = (l + r) / 2;
            res = Math.Min(res, nums[mid]);

            if (nums[mid] >= nums[l])
                l = mid + 1;
            else
                r = mid - 1;
        }

        return res;
    }
}

File: Data Structures & Algorithms/merge-intervals/submission-0.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        # Overlaps if:
        # Start/end point of one interval is the start/end point of another
        # One interval swallows another

        # Are the intervals in "intervals" always sorted?

        # Loop through "intervals". If overlapping criteria is met, take min start and max end of each.
        # Add to resulting list.

        def is_overlapping(interval1: List[int], interval2: List[int]) -> bool:
            return interval1[0] <= interval2[1] and interval2[0] <= interval1[1]

        res = [intervals[0]]
        res_last_index = 0
        i = 1
        while i < len(intervals):
            if is_overlapping(res[res_last_index], intervals[i]):
                min_start = min(res[res_last_index][0], intervals[i][0])
                max_end = max(res[res_last_index][1], intervals[i][1])
                res[res_last_index][0] = min_start
                res[res_last_index][1] = max_end
            else:
                res.append(intervals[i])
                res_last_index += 1
            i += 1

        return res